### PR TITLE
Add new condition types

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Adds three new conditon types to account for common business scenarios.
| Type?             | new feature
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Try to use those new conditions.
| UI Tests          | 
| Fixed issue or discussion?     | https://github.com/PrestaShop/PrestaShop/discussions/35945
| Related PRs       | Autoupgrade PR must come.
| Sponsor company   | 

### New conditions
- Damaged - slight item damage, from shipping for example, but can be sold just fine.
- Open package - items returned in 14 day period, but new apart from having opened packaging.
- B grade - items that have some slight manufacturing defects, but can be sold just fine.

### CODE REVIEW

**Comment by kpodemski** [comment]
Hi @Hlavtox 

Could you create [an idea](https://github.com/PrestaShop/PrestaShop/discussions/categories/ideas) here just for other members of @PrestaShop/product-council to confirm it is valid and get it done?

**Comment by Hlavtox** [comment]
@kpodemski I created an idea there - https://github.com/PrestaShop/PrestaShop/discussions/35945

**Review by ShaiMagal — DISMISSED** [review]

**Comment by jf-viguier** [comment]
it's ok for me

**Comment by Hlavtox** [comment]
@kpodemski As one member of product council approved and no others commented, I am putting the PM flag.

**Comment by MatShir** [comment]
IMO this improvement is specific to your uses cases. 
We should either leave that way and let the user update the DB for their personal use or make it dynamic.

**Review by MatShir — CHANGES_REQUESTED** [review]
IMO this improvement is specific to your uses cases. 
We should either leave that way and let the user update the DB for their personal use or make it dynamic (core or native module)
I don't see why it should be one part of the core project.

**Comment by Hlavtox** [comment]
@MatShir It's not specific to my usecase, its a fairly common request from my clients.
Making it dynamic is much more work. This works and gets the job done for minimal effort, with no BC breaks.

Moreover, googling "product condition types" brings several articles of ecommerce companies. And one of the first results is for example this - https://pinchestimating.com/product-condition-types/, which contains all of my newly added types.

- Open Box: The product is in like-new condition but has been opened or unboxed. It may have been briefly used for demonstration or display purposes, but it should still function perfectly.
- Damaged/For Parts: The product has significant damage or is non-functional. It is being sold primarily for salvageable parts or repair purposes. The description should outline the specific damage or issues.
- Scratch and Dent: The product has cosmetic flaws such as scratches, dents, or minor damage. While the appearance may be affected, the product should still function properly.

**Comment by kpodemski** [comment]
> IMO this improvement is specific to your uses cases.

I also disagree. It's not a specific use case but a set of different, industry-standard conditions added to a selection.

While I agree that this could be configurable, I don't see it as something half-baked, it can be safely implemented since it is backward compatible, and provides extra option for a user.

> I don't see why it should be one part of the core project.

Modifying this part of the core is very tedious work and can't be easily done without messing with a lot of core files.

**Comment by MatShir** [comment]
I agree there is a lack of labels available. There are so many different from depending on the businesses (https://www.amazon.com/gp/help/customer/display.html?nodeId=201889720).
It has never been a big issue on since I think people use features to bypass it. 
So adding new inputs that don't fit all the needs IMO is not worth it.

Or we should a bunch more, so people can change by changing the translation 🤔

**Comment by Hlavtox** [comment]
Nobody will program this to be dynamic, it's just too much work. Creating database tables, providers, controllers, forms. Also, we would need to prevent people deleting the default ones. Also, now it would be possible to map the condition to other modules. Many caveeats for small value.

If somebody has very specific business, he can modify the core more to add his own conditions. These ones are general ones that can be applied to every business. Every merchant in EU sells returned open box items once in a while.

**Comment by MatShir** [comment]
I agree that making a dynamic fields would be quite expensive and not really worth it. I suggest only removing **B grade** which is a repetition from the damaged and opened.

File: src/Adapter/Presenter/Product/ProductLazyArray.php [hunk 1]
```diff
@@ -278,6 +278,24 @@ public function getCondition()
...
+            case 'damaged':
+                return [
+                    'type' => 'damaged',
+                    'label' => $this->translator->trans('Damaged', [], 'Shop.Theme.Catalog'),
+                    'schema_url' => 'https://schema.org/DamagedCondition',
+                ];
+            case 'open_package':
+                return [
+                    'type' => 'open_package',
+                    'label' => $this->translator->trans('Opened package', [], 'Shop.Theme.Catalog'),
```
Comment (Julievrz) [inline src/Adapter/Presenter/Product/ProductLazyArray.php]: You mentionned in the description that the product is still new. We could specify that in the label. For example, Amazona and Ebay use "New - Open box" which I think is very clear.

File: src/Adapter/Presenter/Product/ProductLazyArray.php [hunk 2]
```diff
@@ -278,6 +278,24 @@ public function getCondition()
...
+                    'schema_url' => 'https://schema.org/NewCondition',
+                ];
+            case 'b_grade':
+                return [
+                    'type' => 'b_grade',
+                    'label' => $this->translator->trans('B grade', [], 'Shop.Theme.Catalog'),
```
Comment (Julievrz) [inline src/Adapter/Presenter/Product/ProductLazyArray.php]: What about "New with defects". This way we understand the product is new but could have small imperfections.

**Comment by Julievrz** [comment]
Also I don't understand the difference between "Damaged" and "B-grade", I think we could have only 1 label like "New with defects"

**Comment by Hlavtox** [comment]
@Julievrz
Damaged - dents, scratches from shipping or bad handling
B - grade/Second - items with manufacturing defects, bad paint, misaligned parts that didn't pass inspection as good quality items (https://en.wikipedia.org/wiki/Factory_second)

**Comment by Hlavtox** [comment]
OK, so final list?
- New
- New, open box
- New, minor defects
- Used
- Refurbished
- Damaged

Regarding structured data. For `New, open box` and `New, minor defects` I will use https://schema.org/NewCondition, because schema.org doesnt have definition for them - https://schema.org/OfferItemCondition.

**Comment by MatShir** [comment]
So Damaged is not a new product ? 
Damaged and new with defects/minor defects kind of look very similar. From your description the damages, one comes from the factory and the other from the last shipment, but the defects are the same. So why make a difference for the consumer?

**Comment by Julievrz** [comment]
> @Julievrz Damaged - dents, scratches from shipping or bad handling B - grade/Second - items with manufacturing defects, bad paint, misaligned parts that didn't pass inspection as good quality items (https://en.wikipedia.org/wiki/Factory_second)

I understand but for the final customer, the result seems to be the same: the product is slightly imperfect. Why do we need to have 2 different labels?

Also "Damaged" could be confused with "For parts" or "Not working" when a product is damaged and needs repair.

**Comment by Hlavtox** [comment]
@Julievrz @MatShir The merchants usually always specify what is wrong with the product by photos and descriptions, showing the damage, imperfections. The conditions are only for general classification.

**Comment by Julievrz** [comment]
> @Julievrz @MatShir The merchants usually always specify what is wrong with the product by photos and descriptions, showing the damage, imperfections. The conditions are only for general classification.

I think this is another reason not to get too specific. There are many other conditions we could add but if we want to keep it simple, I think one condition "New, minor defects" could work for both use cases. Especially if there are pictures.

**Comment by Hlavtox** [comment]
@jf-viguier @ShaiMagal @kpodemski Can you state your opinion here?

A)
New
New, open box
New, minor defects
Used
Refurbished
Damaged

or

B)
New
New, open box
New, minor defects
Used
Refurbished

I don't think it's such an issue to have 6 options instead of 5. :-)))

**Review by FabienPapet — DISMISSED** [review]

**Comment by kpodemski** [comment]
@Hlavtox 

New, minor defects to me sound good enough to describe "New, open box"; the details could be in the description.

**Comment by Hlavtox** [comment]
@kpodemski Thats not what was questioned, @Julievrz was asking of we want to merge `Damaged` and `New, minor defects` together.

**Comment by kpodemski** [comment]
@Hlavtox 

ahh, all right, Damaged sounds more serious than New, minor defects 🤔 but then it's about how damaged is minor and major, soooo, I don't know, I feel like we definitely spend too much time on this PR, maybe keeping more options would be easier for end user to decide what they feel is ok for them :)

**Comment by Hlavtox** [comment]
@kpodemski Yeah, it could be merged 3 weeks ago with 1 hour spent...

Users open this box only for items with some non standard condition, and if we want to please more of the merchants, 6 options instead of 5 has no harm. Certainly it won't endanger the business of Prestashop SA. :-)

So, this is the time I want to spend on it.

@MatShir @Julievrz If you disagree, I will just close it.

**Review by kpodemski — APPROVED** [review]

**Comment by Julievrz** [comment]
> @kpodemski Yeah, it could be merged 3 weeks ago with 1 hour spent...
> 
> Users open this box only for items with some non standard condition, and if we want to please more of the merchants, 6 options instead of 5 has no harm. Certainly it won't endanger the business of Prestashop SA. :-)
> 
> So, this is the time I want to spend on it.
> 
> @MatShir @Julievrz If you disagree, I will just close it.

Hello Daniel,

Actually, my recommendation was "New, with defects" (not "minor defects").

In my opinion, the problem is not to have 6 options instead of 5. The problem is to have 2 options that mean the same thing.

**Comment by Hlavtox** [comment]
@Julievrz `Manufacturing defects` (manufacturer second) is not the same thing as `scratch and dent`. Source - my work for 15 years.

**Comment by Hlavtox** [comment]
Closing this as we are unable to find consensus with Prestashop SA employees.

**Pull request closed without merging**